### DISCUSSION TITLE
format for script sensor output

### DISCUSSION THREAD

**Discussion by oldmanbluntz** [theyosh/TerrariumPI#833] [Q&A] [answered]
I'm just curious what format TerrariumPi is expecting in regards to a moisture sensor. I have the script sensor all working, i just can't get terrarium pi to update the script sensor on the GUI for the moisture sensors i have.

They are attached to an arduino. I an sending the data through a structured byte stream over I2C to the raspberry pi. My python scripts work and will output a percentage number when i run the scripts. I just don't know what format terrarium pi expects.

If anyone has some insight on this, i would be eternally happy, and it would be a great help.

**Comment by theyosh** [comment, marked as answer]
Hi, sorry for missing your question at Matrix. 

But the output of the script should be a float/number only. No other text. And no % or C signs.
Depending on the sensor type (moisture in your case) the correct sign will be added. And make sure the values are within the min and max values of the sensor in the GUI.

**Comment by theyosh** [comment]
And the script can be located anywhere on the disk, but I advise to create a 'script' folder in the TerrariumPI folder. That mimics the docker setup.

**Comment by oldmanbluntz** [comment]
okay, so the 0-1023 range and nothing else?
